lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -607,7 +607,8 @@ void TargetPassConfig::addMachinePasses() {
     addPass(&LocalStackSlotAllocationID, false);
   }
 
-  addPass(createPagerandoBinningPass());
+  if (TM->isPagerando())
+    addPass(createPagerandoBinningPass());
```

Comment:
Although PagerandoBinning is target-agnostic, we could push this inside the target-specific `PassConfig`s to have the everything Pagerando right next to each other. What do you think?